adapter/slack.py

```diff
@@ -53,6 +102,82 @@ def send_generated_response(
                 text="Something went wrong when trying to generate your response.",
             )
 
+    async def ask_form(self, _: Request):
+        return {
+            "blocks": [
+                {
+                    "type": "section",
+                    "block_id": "bots",
+                    "text": {
+                        "type": "mrkdwn",
+                        "text": "*Select the chatbot you want to ask*",
+                    },
+                    "accessory": {
+                        "action_id": "select_chatbot",
+                        "type": "external_select",
+                        "placeholder": {
+                            "type": "plain_text",
+                            "text": "Select a chatbot",
+                        },
+                        "min_query_length": 0,
+                    },
+                },
+                {
+                    "type": "input",
+                    "block_id": "question",
+                    "element": {
+                        "action_id": "question",
+                        "type": "plain_text_input",
+                        "multiline": True,
+                        "min_length": 1,
+                    },
+                    "label": {"type": "plain_text", "text": "What is your question?"},
+                    "hint": {
+                        "type": "plain_text",
+                        "text": "e.g. Who's the CEO of our company?",
+                    },
+                },
+                {
+                    "type": "actions",
+                    "elements": [
+                        {
+                            "type": "button",
+                            "text": {"type": "plain_text", "text": "Ask question"},
+                            "style": "primary",
+                            "value": "ask_question",
+                            "action_id": "ask_question",
+                        },
+                    ],
+                },
+            ]
+        }
+
+    async def ask_v2(self, channel_id: str, user_id: str, slug: str, question: str):

```

Comment:
Do you think we can split some code from ask into another method so that ask_v2 wouldn't be necessary?

Reply:
I believe it might be best not to refactor it into separate methods at this point. The current logic seems straightforward, and breaking it down could lead to an excessive number of small functions, which might complicate readability and maintenance rather than enhance it.

Reply:
We probably can but I think it's better to leave that for another day to handle the other tasks